### PR TITLE
[feat]: [CDS-53556]: Proper error handling in case of unsupported workload kinds

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Added proper error handling in case of unsupported workload kinds

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44393)

### CODE REVIEW

File: 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sScaleRequestHandler.java [hunk 1]
```diff
@@ -184,6 +187,15 @@ void init(K8sScaleRequest request, K8sDelegateTaskParams k8sDelegateTaskParams,
       resourceIdToScale.setNamespace(namespace);
     }
 
+    if (!K8S_SCALING_SUPPORTED_WORKLOAD_KINDS.contains(resourceIdToScale.getKind())) {
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sScaleRequestHandler.java:190]: what if kind is lowercase or any other case that the one you expect?

**Review by abosii-harness — CHANGES_REQUESTED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sScaleRequestHandler.java (hunk 1 above)
Comment (lovish1999) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/k8s/K8sScaleRequestHandler.java:190]: Ok.

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger checkstylecodeformat

**Comment by lovish1999** [comment]
trigger messagecheck

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44393)* status: 6 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java` line 1151 at r3](https://reviewable.io/reviews/harness/harness-core/44393#-NPMVgOw04zcG5__TGnp:-NPMVgOw04zcG5__TGnq:bvf70m3) ([raw file](https://github.com/harness/harness-core/blob/77ca371d5600b3f74aabe24e1e83aff380becf14/930-delegate-tasks/src/main/java/io/harness/delegate/task/k8s/K8sTaskHelperBase.java#L1151)):*
> ```Java
>     ProcessResult result = runK8sExecutableSilent(k8sDelegateTaskParams, getCommand);
>     if (result.getExitValue() == 0) {
>       if (isEmpty(result.outputUTF8().trim())) {
> ```

we have to check for !result.hasOutput() 
let's just pass the logcallback here and log a warning: Unable to retrieve current replicas count based on path $.spec.replicas, resource 'x' may not be supported for scaling
and then return null

**Review by namantalaycha10 — APPROVED** [review]
> Reviewed 8 of 9 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44393)* status: 7 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 8 of 9 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44393)* status: :shipit: complete! ✓

**Comment by lovish1999** [comment]
trigger ut5

**Comment by lovish1999** [comment]
trigger ut6

**Comment by lovish1999** [comment]
trigger ut7

**Comment by lovish1999** [comment]
trigger ut8

**Comment by lovish1999** [comment]
trigger ut9

**Review by TarunUba1 — APPROVED** [review]
> Reviewed 8 of 9 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44393)* status: :shipit: complete! ✓

**Comment by lovish1999** [comment]
trigger ut6

**Pull request merged**